src/components/shared/AppBar/AppBar.js

```diff
@@ -0,0 +1,86 @@
+import React, {useEffect, useState} from 'react';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import InputBase from '@material-ui/core/InputBase';
+import SearchIcon from '@material-ui/icons/Search';
+import {Button, ButtonGroup} from "@material-ui/core";
+import {useDispatch} from "react-redux";
+import {loadBreeds} from "../../../store/actions/breedsReducerActions";
+import {useHistory, BrowserRouter as Router} from "react-router-dom";
+import {inputAction} from "../../../store/actions/inputAction";
+import {useStyles} from "./AppBarStyles";
+
+export function SearchAppBar() {
+    const classes = useStyles();
+    const dispatch = useDispatch();
+    useEffect(() => {
+        dispatch(loadBreeds());
+    }, [dispatch])
+
+    const [inputState, setInputState] = useState('');
```

Comment:
Do you really need this input state? You can use input from redux, and update it using dispatch